src/framework/components/script/component.js

```diff
@@ -191,7 +202,7 @@ class ScriptComponent extends Component {
      * An array of all script instances attached to an entity. This array is read-only and should
      * not be modified by developer.
      *
-     * @type {import('../../script/script-type.js').ScriptType[]}
+     * @type {import('../../script/script.js').Script[]}
```

Comment:
not related to this PR directly, but the function says it's read only .. so perhaps we should not have a setter at all? Only a public getter.
And this setter could be changed to something like `setScripts` which is `@ignore`

Reply:
can look at this in followup PR